### DISCUSSION TITLE
Theos

### DISCUSSION THREAD

**Discussion by ExtraBytez** [godotengine/godot-proposals#10363] [Platforms]
could godot somehow use theos https://theos.dev/ to build to ios from windows?

**Comment by Calinou** [comment]
Using Theos would require replacing the entire SCons buildsystem with it. Given how complex Godot's usage of SCons is, this is not technically viable.

Also, Theos seems to cater to jailbroken iOS devices first and foremost. It's unclear whether you can submit apps compiled with it to the iOS App Store.

**Reply by ExtraBytez** [reply]
fair guess not everyone will have jb + most people who compile for ios have a mac

**Comment by bruvzg** [comment]
Cross-building iOS export template (static library) is already supported with [OSXCross](https://docs.godotengine.org/en/latest/contributing/development/compiling/cross-compiling_for_ios_on_linux.html) (that's how official templates are built), on Windows you can use WSL / Podman / VM to build it.

The export process for iOS currently works like this:
- Godot extract static library and Xcode project for the export template.
- Project metadata is changed according to the project/export settings.
- Packed resources and other static/dynamic libraries (GDExtension or .NET) are added to the project.
- *[everything up to this point can be done on any platform]*
- Godot runs `xcodebuild` to build the project.

Since Theos doesn't provide `xcodebuild` equivalent (neither do OSXCross), it's won't help with this step.
